File: Common/Serialization/FormatterSetProxy.vb
Public Class FormatterSetProxy

    Public Sub New(ByVal Formatter As Formatter)
        Me._Formatter = Formatter
    End Sub

    Protected Friend Sub [Set](Of T)(ByVal Name As String, ByVal Obj As T)
        Me.Formatter.Set(Name, Obj)
    End Sub

    Protected Friend Sub [Set](ByVal Name As String, ByVal Obj As Object)
        Me.Formatter.Set(Name, Obj)
    End Sub

#Region "Formatter Property"
    Private ReadOnly _Formatter As Formatter

    Public ReadOnly Property Formatter As Formatter
        Get
            Return Me._Formatter
        End Get
    End Property
#End Region

End Class

File: Common/Serialization/FormatterSetProxy.cs
namespace Ks.Common
{
    public class FormatterSetProxy
    {
        public FormatterSetProxy(Formatter Formatter)
        {
            this._Formatter = Formatter;
        }

        protected internal void Set<T>(string Name, T Obj)
        {
            this.Formatter.Set(Name, Obj);
        }

        protected internal void Set(string Name, object Obj)
        {
            this.Formatter.Set(Name, Obj);
        }

        private readonly Formatter _Formatter;

        public Formatter Formatter
        {
            get
            {
                return this._Formatter;
            }
        }
    }
}
